lumigator/backend/backend/services/jobs.py

```diff
@@ -57,25 +60,167 @@
 JobSpecificRestrictedConfig = type[JobEvalConfig | JobInferenceConfig]
 
 
+# The end result should be that InferenceJobConfig is actually JobInferenceConfig
+# (resp. Eval)
+# For the moment, something will convert one into the other, and we'll decide where
+# to put this. The jobs should ideally have no dependency towards the backend.
+
+
+class JobDefinition(ABC):
+    """Abstract base class for jobs.
```

Comment:
@peteski22  recently refactored my tracking_interface code to use protocols instead of ABC: Peter, is that relevant here?

Reply:
We'll stick to abstract classes for the moment. I'm really divided in this respect, but we can retrofit this if needed. Would _you_ prefer abstract classes or plain protocols?